server/game/cards/01_SOR/events/Restock.ts

```diff
@@ -0,0 +1,39 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { RelativePlayer, TargetMode, ZoneName } from '../../../core/Constants';
+
+export default class Restock extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '6087834273',
+            internalName: 'restock'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Choose up to 4 cards in a discard pile. Put them on the bottom of their owner\'s deck in a random order',
+            targetResolver: {
+                mode: TargetMode.Select,
+                choices: {
+                    ['Your discard']: AbilityHelper.immediateEffects.selectCard({
+                        mode: TargetMode.UpTo,
+                        numCards: 4,
+                        zoneFilter: ZoneName.Discard,
+                        controller: RelativePlayer.Self,
+                        innerSystem: AbilityHelper.immediateEffects.moveToBottomOfDeck()
+                    }),
+                    ['Opponent discard']: AbilityHelper.immediateEffects.selectCard({
+                        mode: TargetMode.UpTo,
+                        numCards: 4,
+                        zoneFilter: ZoneName.Discard,
+                        controller: RelativePlayer.Opponent,
+                        innerSystem: AbilityHelper.immediateEffects.moveToBottomOfDeck()
```

Comment:
I'll raise this with Caz for FE stuff at some point soon, but I wonder if it would be good to have an extra option for this that makes it so that the player can see the order in which they selected cards. That way they can easily remember what order they'll be put in the bottom of the deck.

Reply:
Restock need to be at random order

Reply:
LOL yep thanks